ZIO2 core metrics interceptor/collector

ianlintner opened the pull request:
ZIO2 Integrated ZMX metrics into core with v2 adding an interceptor/collector makes sense.

* Did not include a metric endpoint as zio metrics don't define output only collection.

adamw commented on build.sbt:
```diff
@@ -904,6 +904,20 @@ lazy val datadogMetrics: ProjectMatrix = (projectMatrix in file("metrics/datadog
   .jvmPlatform(scalaVersions = scala2And3Versions)
   .dependsOn(serverCore % CompileAndTest)
 
+lazy val zioMetrics: ProjectMatrix = (projectMatrix in file("metrics/zio-metrics"))
```
also needs to be added to aggregates so that publishing works correctly

adamw commented on metrics/zio-metrics/src/main/scala/sttp/tapir/server/metrics/zio/ZioMetrics.scala:
```diff
@@ -0,0 +1,178 @@
+}
+
+object ZioMetrics {
+  val DEFAULT_NAMESPACE: String = "tapir"
```
[minor] by convention constants in scala are camel case, e.g. `DefaultNamespace`

adamw commented:
Thanks! I think this should also be mentioned in the observability docs: https://tapir.softwaremill.com/en/latest/server/observability.html so that people can discover the module

ianlintner commented:
I incorporated your suggestions and changes. 
* Added comments & cleanup.
* Added documentation to observability.md
* Also added an example in example project similar to other metric collectors.
* Got a clean build in scala3 CI as well.

adamw commented:
Thanks! :)
